### ISSUE TITLE
Why does the hook return a mutable value?

### ISSUE THREAD

**Issue by TrueWill** [immerjs/use-immer#85]
```ts
export function useImmer(initialValue: any) {
  const [val, updateValue] = useState(initialValue);
  return [
    val,
```

I was really surprised that this didn't return `freeze(val, true)` or similar. This seems to defeat the purpose of insuring immutability to me. What's the thinking behind this? Thanks!

**Comment by mweststrate** [comment]
I don't understand the question. Please create a CodeSandbox that explains why you are observing and what you'd expect

**Comment by TrueWill** [comment]
@mweststrate Thanks for your patience.

Please see https://codesandbox.io/s/use-immer-mutable-bfeoi

I made it a little more complicated to reflect what I'm trying to accomplish. I want to create a React context with a value that can only be mutated through setters.

If you start with a fresh page and click "Change it (good)", that will update the context value object through the setter with a draft. That's what I want.

If you refresh the page and click "Change it (BAD)", the console will show "Just mutated state!" That's what I'd like to avoid - components accidentally mutating the context value object directly.

To show how I'd expect this to work, you can open fooContext.js and uncomment the two lines (commenting out `return context;`). Now the initial object is frozen, and clicking the "BAD" button won't change anything.

If I'm missing something, or if there's a better way to implement this, please share.

Thank you!

**Comment by mweststrate** [comment]
Ah, check, thanks for clarifying and spotting. The reason the object is not frozen is that the initial state is never processed by Immer, but stored directly by useState here https://github.com/immerjs/use-immer/blob/master/src/index.ts at line 14. To have always frozen state line 14 and 19 should be updated to freeze the newly stored state. Feel free to create a PR for that!

**Comment by mweststrate** [comment]
Should be fixed in 0.6.0

**Comment by TrueWill** [comment]
@mweststrate Thank you so much! Your libraries have been a big help. I apologize for dropping the ball and not creating a PR.
